### PR TITLE
upload assets on published release

### PR DESCRIPTION
**Pull request by jetersen**
Noted the last two releases did not come with all the assets, so created this to upload them all on publish.

### CODE REVIEW

**Comment by jetersen** [comment]
Not sure what you use the `updateCli.env` for 🤔

File: .github/workflows/publish-assets.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Build updateCli
+      env:
+        TAG: ${{ github.ref }}
```
Comment (jetersen) [inline .github/workflows/publish-assets.yaml:18]:
```suggestion
```

taken from another example 😓

File: utils/build.sh [hunk 2]
```diff
@@ -8,11 +8,11 @@ BASENAME="updatecli"
...
 echo "Building updatecli OS packages for version $VERSION"
 
-for GOOS in darwin linux windows; do 
-  for GOARCH in amd64 386; do 
+for GOOS in darwin linux windows; do
+  for GOARCH in amd64 386; do
 
     BINARY="$BASENAME.$GOOS.$GOARCH"
-    BINARY_DIRECTORY="./bin/"
+    BINARY_DIRECTORY="./bin"
```
Comment (jetersen) [inline utils/build.sh:15]: The path had double slashes

see line 23 and 30.

File: .github/workflows/publish-assets.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
+name: Upload assets to releases
+
+on:
+  release:
+    types:
+      - created
```
Comment (jetersen) [inline .github/workflows/publish-assets.yaml:6]:
```suggestion
      - published
```

**Comment by jetersen** [comment]
We have access to all this event data: https://docs.github.com/en/free-pro-team@latest/developers/webhooks-and-events/webhook-events-and-payloads#release

File: .github/workflows/publish-assets.yaml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: Build updateCli
+      run: |
+        touch updateCli.env
+        make build.all
```
Comment (olblak) [inline .github/workflows/publish-assets.yaml:19]: Does the ubuntu-latest environment has golang installed?

Comment (olblak) [reply .github/workflows/publish-assets.yaml:19]: Apparently yes - [doc](https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md#go)

**Review by olblak — APPROVED** [review]

**Comment by olblak** [comment]
Thanks, awesome, I wanted to work on this for a while

> Not sure what you use the `updateCli.env` for thinking

I use it source local environment variables, it's not needed anymore with your PR, I'll remove it

**Comment by jetersen** [comment]
Your very welcome!

**Pull request merged**